File: Codes/1742-maximum-number-of-balls-in-a-box_20250213.go
/*
 * @Author: LetMeFly
 * @Date: 2025-02-13 16:22:21
 * @LastEditors: LetMeFly.xyz
 * @LastEditTime: 2025-02-13 16:31:13
 */
package main

func get_MNB1B(n int) (ans int) {
    for n > 0 {
        ans += n % 10
        n /= 10
    }
    return
}

func min_MNB1B(a, b int) int {
    if a < b {
        return a
    }
    return b
}

func countBalls(l int, h int) (ans int) {
    cnt := make([]int, 46)
    for ; l <= h; l++ {
        thisCnt := get_MNB1B(l)
        cnt[thisCnt]++
        ans = max(ans, cnt[thisCnt])
    }
    return
}

File: Codes/1742-maximum-number-of-balls-in-a-box_20250213.java
/*
 * @Author: LetMeFly
 * @Date: 2025-02-13 16:22:18
 * @LastEditors: LetMeFly.xyz
 * @LastEditTime: 2025-02-13 16:27:51
 */
class Solution {
    private int get(int n) {
        int ans = 0;
        while (n > 0) {
            ans += n % 10;
            n /= 10;
        }
        return ans;
    }
    public int countBalls(int l, int r) {
        int[] cnt = new int[46];
        int ans = 0;
        for (; l <= r; l++) {
            int thisCnt = get(l);
            cnt[thisCnt]++;
            ans = Math.max(ans, cnt[thisCnt]);
        }
        return ans;
    }
}